### PR TITLE
fix: Balance channel may cause channel not availble error

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #28831
release old delegator before new delegator update it's distribution may cause `channel not availble` error

This PR will block release old delgator before new delegator finish `syncDistribution`

### CODE REVIEW

File: internal/querycoordv2/task/scheduler.go [hunk 1]
```diff
@@ -614,12 +614,52 @@ func (scheduler *taskScheduler) isRelated(task Task, node int64) bool {
...
+		zap.Int64("collectionID", task.CollectionID()),
+		zap.Int64("taskID", task.ID()),
+	)
 	if task.Status() != TaskStatusStarted {
 		return false
 	}
 
+	// check if new delegator is ready to release old delegator
+	checkLeaderView := func(collectionID int64, channel string, node int64) bool {
+		segmentsInTarget := scheduler.targetMgr.GetSealedSegmentsByChannel(collectionID, channel, meta.CurrentTarget)
```
Comment (yah01) [inline internal/querycoordv2/task/scheduler.go:627]: Move these into the channel task's `IsSucceeded` method

**Review by yah01 — CHANGES_REQUESTED** [review]

**Review by yah01 — APPROVED** [review]
/lgtm

**Review by yah01 — COMMENTED** [review]
/lgtm

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**